pkg/util/net.go

```diff
@@ -34,16 +34,33 @@ import (
 
 var previousGatewayIP string = ""
 
-func GetHostIP() (string, error) {
-	ip, err := net.ChooseHostInterface()
-	if err == nil {
-		return ip.String(), nil
+func GetHostIP(nodeIP string) (string, error) {
+	var hostIP string
+	var err error
+
+	if nodeIP != "" {
+		klog.Infof("try to find nodeIP %s on host", nodeIP)
+		hostIP, err = selectV4IPFromHostInterface(nodeIP)
+		if err != nil {
+			return "", fmt.Errorf("failed to find nodeIP %s on host: %v", nodeIP, err)
+		}
+		goto found
```

Comment:
Oh, my, a goto?!  ;-)

That probably could have been done with a defer, but this is clear, too.

Reply:
I used to use an `if / else` here. But it cannot pass our nestedif check. So...